### PR TITLE
code-refactoring: Gitops 3127 implement applicationset controller package

### PR DESCRIPTION
**Pull request by ciiay**
**What type of PR is this?**
/kind code-refactoring

**What does this PR do / why we need it**:
Implemented notifications controller.

NOTE: this PR just adds the package files and tests, this code is not invoked at any point yet
**Have you updated the necessary documentation?**

* [ ] Documentation update is required by this PR.
* [ ] Documentation has been updated.

**Which issue(s) this PR fixes**:

Fixes #?
GITOPS-3127: https://issues.redhat.com/browse/GITOPS-3127

**How to test changes / Special notes to the reviewer**:
Unit tests and kuttl tests uploaded.

### CODE REVIEW

File: controllers/argocd/applicationset/constants.go [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package applicationset
+
+// Values
+const (
+	ArgoCDApplicationSetControllerComponent = "applicationset-controller"
+	ArgoCDApplicationSetController          = "argocd-applicationset-controller"
+	ApplicationSetGitlabSCMTlsCert          = "appset-gitlab-scm-tls-cert"
+	ApplicationSetGitlabSCMTlsCertPath      = "/app/tls/scm/cert"
```
Comment (jaideepr97) [inline controllers/argocd/applicationset/constants.go:8]: can we make this uniform? 
`AppSetControllerComponent`, `AppSetGitlabSCMTlsCert` etc

File: controllers/argocd/applicationset/deployment.go [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+	}
+	asr.Logger.V(0).Info("DeleteDeployment: deployment deleted", "name", name, "namespace", namespace)
+	return nil
+}
+
+func (asr *ApplicationSetReconciler) getArgoApplicationSetCommand() []string {
+	cmd := make([]string, 0)
+
+	cmd = append(cmd, common.EntryPointSh)
+	cmd = append(cmd, ArgoCDApplicationSetController)
```
Comment (jaideepr97) [inline controllers/argocd/applicationset/deployment.go:180]: probably better to rename this variable as `AppSetBinaryName`

File: controllers/argocd/applicationset/role.go [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+	if err := permissions.DeleteRole(name, namespace, asr.Client); err != nil {
+		asr.Logger.Error(err, "DeleteRole: failed to delete role", "name", name, "namespace", namespace)
+		return err
+	}
+	asr.Logger.V(0).Info("DeleteRole: role deleted", "name", name, "namespace", namespace)
+	return nil
+}
+
+func getPolicyRules() []rbacv1.PolicyRule {
+	return []rbacv1.PolicyRule{
```
Comment (jaideepr97) [inline controllers/argocd/applicationset/role.go:91]: @ciiay can we update these permissions to match https://github.com/argoproj/argo-cd/blob/master/manifests/install.yaml#L20351?

File: controllers/argocd/applicationset/rolebinding.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+)
+
+func (asr *ApplicationSetReconciler) reconcileRoleBinding() error {
+
+	asr.Logger.Info("reconciling roleBindings")
+
+	sa, err := permissions.GetServiceAccount(resourceName, asr.Instance.Namespace, asr.Client)
+
+	if err != nil {
+		asr.Logger.Error(err, "reconsileRoleBinding: failed to get serviceaccount", "name", resourceName, "namespace", asr.Instance.Namespace)
```
Comment (jaideepr97) [inline controllers/argocd/applicationset/rolebinding.go:22]: nit: typo in fn name

File: controllers/argocd/notifications/serviceaccount_test.go
Comment (jaideepr97) [inline controllers/argocd/notifications/serviceaccount_test.go:1]: why are these changes included in this PR?

File: common/cmds.go [hunk 5]
```diff
@@ -0,0 +1,7 @@
+package common
+
+const (
+	EntryPointSh     = "entrypoint.sh"
+	LogLevel         = "--loglevel"
+	ArgoCDRepoServer = "--argocd-repo-server"
```
Comment (jaideepr97) [inline common/cmds.go:6]: Can we move this to the repo-server package?

**Comment by jaideepr97** [comment]
@ciiay if all the old code has been accounted for, please remove those files 
`controllers/argocd/applicationset.go` and `controllers/argocd/applicationset_test.go`

File: controllers/argocd/applicationset/constants.go (hunk 1 above)
Comment (ciiay) [reply controllers/argocd/applicationset/constants.go:8]: In `notifications` controller, they are 
```
	ArgoCDNotificationsControllerComponent = "notifications-controller"
	NotificationsSecretName                = "argocd-notifications-secret"
	NotificationsConfigMapName             = "argocd-notifications-cm"
```
So I followed the same convention.

Do you want them changed to the below names?
```
NotificationsControllerComponent
NotificationsSecretName
NotificationsConfigMapName

AppSetControllerComponent
AppSetController
AppSetGitlabSCMTlsCert
AppSetGitlabSCMTlsCertPath
```

File: controllers/argocd/applicationset/role.go (hunk 3 above)
Comment (ciiay) [reply controllers/argocd/applicationset/role.go:91]: Sure. Then this means the current roles in applicationset controller doesn't match the manifests you provided, and it was okay?

Comment (jaideepr97) [reply controllers/argocd/applicationset/role.go:91]: yeah its fine
maybe we just didnt update them
this seems like a good chance to get aligned again

File: controllers/argocd/applicationset/constants.go (hunk 1 above)
Comment (jaideepr97) [reply controllers/argocd/applicationset/constants.go:8]: yeah, I dont think we need ArgoCD in every constant 
we can reserve that for the general constants, and component specific constants can just start with the component name

File: controllers/argocd/applicationset/role.go (hunk 3 above)
Comment (ciiay) [reply controllers/argocd/applicationset/role.go:91]: Cool, as long as you are sure the manifest ones are more up to date and reliable.

File: controllers/argocd/applicationset/constants.go [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+	// Values
+	AppSetControllerComponent  = "applicationset-controller"
+	AppSetController           = "argocd-applicationset-controller"
+	AppSetGitlabSCMTlsCert     = "appset-gitlab-scm-tls-cert"
+	AppSetGitlabSCMTlsCertPath = "/app/tls/scm/cert"
+
+	// Commands
+	EntryPointSh     = "entrypoint.sh"
+	LogLevel         = "--loglevel"
+	ArgoCDRepoServer = "--argocd-repo-server"
```
Comment (jaideepr97) [inline controllers/argocd/applicationset/constants.go:13]: @ciiay LogLevel should be in the `common` package and `ArgoCDRepoServer` should be in the repo server package
